### PR TITLE
Dependency updates for v2022.4 + cleaning

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I've ran automated tests and reviewed implemented chnages.

#### Why is this the best possible solution? Were any other approaches considered?
As usual I've updated dependencies for next release. This time I've additionally added some cleaning removing unused dependencies and rxjava which was used only in one place (`READ_PHONE_STATE` permission in forms) so I've replaced it with `livedata`.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
This is a kind of change we usually do not test explicitly but merge and do that on a daily basis but this time I've added some changes that we should test first:
- verify that scanning QR codes works well especially on older devices (Android 5)
- verify that requesting `READ_PHONE_STATE` permission works as before in forms. This Permission is required if a from contains [phonenumber](https://docs.getodk.org/form-question-types/#metadata) hidden question. In such a case when a form is started a user should be asked for granting that permission.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java [hunk 1]
```diff
@@ -134,7 +133,7 @@ public List<String> getProperty(String propertyName) {
     @Override
     public String getSingularProperty(String propertyName) {
         if (!permissionsProvider.isReadPhoneStatePermissionGranted() && isPropertyDangerous(propertyName)) {
-            eventBus.post(new ReadPhoneStatePermissionRxEvent());
+            isPhoneNumberRequired.postValue(true);
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java:138]: Could this be `setValue` or is `getSingularProperty` called from background (non-UI) threads?

File: collect_app/src/main/java/org/odk/collect/android/adapters/InstanceUploaderAdapter.java [hunk 2]
```diff
@@ -17,24 +17,13 @@
 
 import java.util.Date;
 
-import io.reactivex.disposables.CompositeDisposable;
-
 import static org.odk.collect.forms.instances.Instance.STATUS_SUBMISSION_FAILED;
 import static org.odk.collect.forms.instances.Instance.STATUS_SUBMITTED;
 
 public class InstanceUploaderAdapter extends CursorAdapter {
-    private final CompositeDisposable compositeDisposable;
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/adapters/InstanceUploaderAdapter.java:26]: Was this just here and not being used for anything?

**Review by seadowg — APPROVED** [review]

**Comment by seadowg** [comment]
🙏 Praise be to @grzesiek2010 for removing RxJava. I think you get to update `STATE.md` as well here.

File: collect_app/src/main/java/org/odk/collect/android/adapters/InstanceUploaderAdapter.java (hunk 2 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/adapters/InstanceUploaderAdapter.java:26]: Yes

File: collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/logic/PropertyManager.java:138]: Yes it's used from background so it needs to be `postValue`

**Comment by grzesiek2010** [comment]
> I think you get to update STATE.md as well here.

Right, done.

**Comment by dbemke** [comment]
**Problem description**

After allowing ODK Collect to make and manage phone calls, opening another form results in continuous loading form state.

**Steps to reproduce the problem**

1. Go to the Metadata form.
[metadata.xml.txt](https://github.com/getodk/collect/files/9394548/metadata.xml.txt)

2. Allow ODK Collect to make and manage phone calls.
3. Exit the form (e.g click device back button and ignore changes).
4. Go to any other form (e.g. All widgets)

**Expected behavior**

Another form should open. The user should be able to complete the form.

**Other information** 

If the user minimizes the app (while loading form is on) and goes back to the form, the form works as expected.
On master allowing to make and manage phone calls doesn't result in continuous loading of forms, but it's possible to get this result by denying ODK Collect making and managing phone calls and in next dialog opening settings, allowing access to phone and going back to the form.

**Comment by grzesiek2010** [comment]
The bug should be fixed. If there is anything else but it exists on master too please file a separate issue.

**Comment by srujner** [comment]
**Tested with Success!**
Verified on Androids: 5.1, 10 and 12

Verified cases:

- Adding project through QR code; 
- Switching between projects, adding duplicated projects; 
- Update settings from older version; 
- Invalid/Inverted QR code;
- Scanning barcodes and QR code with front and back camera;

- Issue found by @dbemke is no longer reproducing;
- Requesting Phone permission dialog;
- Allowing and Denying permissions in the ODK app and in the device settings;
- Screen rotation, minimize app, using backward button;
- Light and Dark theme.

**Comment by dbemke** [comment]
**Tested with Success!**
Verified on Androids: 8.1, 11

**Pull request merged**